### ISSUE TITLE
Variables used in POST requests not properly being substituted in request body

### ISSUE THREAD

**Issue by techpeace** [a85/POSTMan-Chrome-Extension#4718]
**Describe the bug**
I am opening an issue that is the same as that described in #2627, as that bug has still been present for over a year but was closed.

When variables are included in JSON request bodies, they aren't being properly transformed into the correct data type: for example, strings aren't quoted. If I have a variable that is meant to be a string, then it needs to be enclosed in quotation marks in the environment itself or it will come out as a JSON error, as it gets embedded into the JSON body without quotation marks:

```json
{
  "userId": someUserId
}
```

If I need to use that user ID in a URL segment, then I need a version of it that isn't enclosed in quotation marks. The workaround here seems to be creating two environment variables, one including quotation marks, and one without.

**To Reproduce**
1. Add a variable `userId` to your environment with value `someId`.
2. Add a JSON request body like the following:

```json
{
    "userId": {{userId}}
}
```

3. Export a cURL version of the request

```bash
curl -X POST \
  http://example.com/users \
  -H 'Cache-Control: no-cache' \
  -H 'Content-Type: application/json' \
  -H 'Postman-Token: fd735027-af6e-47dd-a61f-bd2f89362ade' \
  -d '{
	"userId": someId
}'
```

**Expected behavior**
The `userId` variable should be enclosed in quotation marks.

**Screenshots**
![image](http://share.voxable.io/1n1E2I2M3y3w/Screen%20Recording%202018-06-19%20at%2010.42%20AM.gif)

**App information (please complete the following information):**
 - App Type: Native app
 - Postman Version: 6.1.3
 - OS: macOS High Sierra 10.13.4

**Additional context**
The problem is that I may need to use this same `userId` in the URL, in which case, I don't want to store it in quotation marks in my environment, as that means it has to be duplicated. That's the best workaround for this issue for now, though.

**Comment by harryi3t** [comment]
Put the variable in quotes.
```js
{
  "userId": "{{userId}}"
}
```

**Comment by techpeace** [comment]
Oh, nice! Yes indeed, that works. It didn't when I originally encountered the issue, and I failed to try that again before I reported.

Thanks!

**Comment by Haveanawesomeday** [comment]
Using double quotes around the variable name doesn't seem to work for me. When I do I receive {{variablename}} instead of its actual value. I am encountering the issue with a JSON (application/json) format and a body with more than one level so I can't use form-data as has been suggested in some threads. 
Running version 6.4.1 on Mac.

**Comment by sujayvenaik** [comment]
Hey @Haveanawesomeday 
Can you send the request that you are trying to send?

I was not able to replicate this, and the variables are getting resolved for all body types.

**Comment by Haveanawesomeday** [comment]
I am seeing this with both the Postman app and the runner, here is the request body below. The variable {{phrase}} is resolved only without double quotes. When I add them it stays "{{phrase}}"
{
  "idOperator": "{{idOperator}}",
  "message": {
    "idMessage": "{{$guid}}",
    "payload": {
      "contentType": "text",
      "value": {{phrase}}
    },
  }
}

**Comment by sujayvenaik** [comment]
Hey @Haveanawesomeday 

I created a dummy request with the same payload as you specified and all values got resolved. You can see the screenshot below.

<img width="1366" alt="screen shot 2018-10-05 at 11 07 25 pm" src="https://user-images.githubusercontent.com/11532114/46550757-98257580-c8f3-11e8-8693-207e8144455d.png">


Can you check once whether you have selected the correct environment in which you have the values?

Also, if you still face the same issues can you send a screen recording with the actual steps you are taking up so we could debug it better.

**Comment by Haveanawesomeday** [comment]
Thank you for the screenshots. I am selecting the env with the right variables. The one causing me trouble, {{phrase}} is the one I extract from the JSON data file in the runner. Is it because all variables come either from an env OR from data files but not both? SHould I specify all env variables in the data file as new keys even if they do not change from one iteration to the next ?

HEre are screenshots below of the request body and data file:
![screen shot 2018-10-08 at 16 34 32](https://user-images.githubusercontent.com/1934600/46615704-3181a100-cb19-11e8-819c-ec6548b9bc39.png)
![screen shot 2018-10-08 at 16 35 19](https://user-images.githubusercontent.com/1934600/46615713-38a8af00-cb19-11e8-96fb-de2d1569c876.png)

**Comment by rafael-yure** [comment]
Yeah I'm having the same issue. The value isn't been replaced.

**Comment by keke9036** [comment]
json { "data":"{{data}}"}  not working for desktop version 6.7.3 ,

**Comment by zehawki** [comment]
Hi, I can confirm that its finally working for me. I'm on v6.7.4 desktop. I've tried 2 cases:

    "recipient_emails": [ "{{email}}" ],
    "receiver_name": "{{name}}",

and my csv has no quotes, escape characters etc

email, name
a@b.com,googoo doll

PS: This has been a frustrating experience since postman prolly believes in incomprehensible documentation as a way to make the community robust. If it were not for this (and #2627) bug details from others, I would not have gotten this working. I've wasted more than an hour trying various methods, with/without quotes, double quotes, escape and what not and trawling thru endless search results. Sigh.
